examples/data-objects/codemirror/src/index.ts

```diff
@@ -15,41 +14,41 @@ import {
     buildRuntimeRequestHandler,
 } from "@fluidframework/request-handler";
 import { defaultRouteRequestHandler } from "@fluidframework/aqueduct";
-
+import { RuntimeFactoryHelper } from "@fluidframework/runtime-utils";
 import { fluidExport as smde } from "./codemirror";
 
-class CodeMirrorFactory implements IRuntimeFactory {
-    public get IRuntimeFactory() { return this; }
+const defaultComponentId = "default";
+const defaultComponent = "@fluid-example/smde";
+
+class CodeMirrorFactory extends RuntimeFactoryHelper {
+    public async instantiateFirstTime(runtime: ContainerRuntime): Promise<void> {
+        await runtime.createRootDataStore(defaultComponent, defaultComponentId);
+    }
 
-    public async instantiateRuntime(context: IContainerContext): Promise<IRuntime> {
+    public async preInitialize(
+        context: IContainerContext,
+        existing: boolean,
+    ): Promise<IRuntime & ContainerRuntime> {
```

Comment:
For readability, I think you can drop IRuntime as ContainerRuntime implements all interfaces that it needs